app/controllers/application_controller.rb

```diff
@@ -88,6 +89,21 @@ def sublayout
 
   protected
 
+  def check_browser
+    return true if request.format.xml? || request.format.json?
+
+    if current_account && current_account.provider_can_use?(:modern_browser_check) && browser_not_modern?
+      logout_keeping_session!
+      flash.now[:error] = 'To protect you, we cannot let you use the browser you are using. Please upgrade your browser and login in again.'
+      redirect_to provider_admin_path
+    end
+  end
+
+  def browser_not_modern?
+    browser = Browser.new(request.env['HTTP_USER_AGENT'])
+    browser.known? && !browser.modern?
```

Comment:
not sure about this, shall we logout user if the browser is unknown?

Reply:
Probably not :)

Reply:
agree.

Reply:
Where are we configuring the rules for the modern browser compatibility? Right now it looks like we are just using the defaults from the library which are not sane at all. Look at the firefox, opera & IE defaults, they are years old. If we are setting this somewhere else where those versions are relevant then ignore this comment and please point me to that commit. We should list this in our supported configurations.

Reply:
Maybe we can use [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options#Browser_compatibility) as a reference?

Reply:
@kevprice83 defaults from the library mirror X-frame-Options support, taken from: https://caniuse.com/#feat=x-frame-options

Reply:
Okay makes sense, the only one that maybe needs review is firefox as the default in the browser gem is [17](https://github.com/fnando/browser/blob/v2.6.1/lib/browser/browser.rb#L87) but in https://caniuse.com/#feat=x-frame-options it is 18. Once we have the final list then I'll speak with Andreu to ensure our supported browser list is up-to-date. Thanks @macejmic